woocommerce/woocommerce
# Migrate E2E daily shopper tests

### All Submissions:


### Changes proposed in this Pull Request:

This PR:
1. Updates `shopper` E2E tests to become runnable and passable in the nightly smoke test site.
1. Fixes remaining flakiness in `merchant` E2E tests.

Since this PR contains many changes, I added PR comments to help explain them.

Closes https://github.com/woocommerce/woocommerce-quality/issues/425.

### How to test the changes in this Pull Request:

1. The daily smoke tests start at 3:25 UTC and runs for about half an hour, so make sure to test this PR outside of this window.
2. Go to https://github.com/woocommerce/woocommerce/actions/workflows/smoke-test-daily.yml
3. Click _Run workflow_.
1. Select this branch (`e2e/daily-3`).
1. Click "Run workflow" button to start running the "Smoke test daily" workflow in this branch.
1. Wait for the workflow to complete.
1. Verify that all the jobs pass.
    <img width="988" alt="Screen Shot 2022-12-09 at 12 18 29 AM" src="https://user-images.githubusercontent.com/4509348/206502793-b217ed90-90ff-485a-87af-7d760a724767.png">
1. Verify that you can see the GitHub Job summary, and that all the links there are working:
    <img width="914" alt="Screen Shot 2022-12-08 at 8 52 31 PM" src="https://user-images.githubusercontent.com/4509348/206503229-acda339b-bbf0-4b60-a107-a8984e34827b.png">
1. Verify that a summary of the report was posted by Woo Test Reporter on Slack. For example: p1670516246559309-slack-C03H038EV2R.
2. Verify that the Allure report was generated, and that the title shows the date and time you ran the test.
    <img width="1028" alt="Screen Shot 2022-12-08 at 8 56 14 PM" src="https://user-images.githubusercontent.com/4509348/206504938-1749781f-2ba0-454b-993e-1ac9d656ca90.png">
3. For @woocommerce/masamune: Verify that all E2E tests pass in the CD.


### Other information:


### FOR PR REVIEWER ONLY:
